src/lib/summary.ts

```diff
@@ -2,51 +2,126 @@ import fs from "fs";
 import { Matches } from "./comparison";
 import { Range } from "./range";
 export type RangesTuple = [Range, Range];
+
+/**
+ * @param minimumLinesInLargestPassage The minimum amount of lines required by the largest code passage. The default
+ * is 1.
+ * @param minimumLinesInSmallestPassage The minimum amount of lines required by the smallest code passage. The default
+ * is 0.
+ * @param passageOutputLimit The maximum amount of code passages that will appear in the output. Everything is shown
+ * if it is anything other then a number greater then zero.
+ */
+export interface FilterOptions {
+  passageOutputLimit?: number;
+  minimumLinesInLargestPassage?: number;
+  minimumLinesInSmallestPassage?: number;
+}
+
 export class Summary {
+  public readonly gapSize: number;
+  public readonly comment: string | undefined;
```

Comment:
I still don't think the comment has to be passed onto the summary. The comment is something output related and has no place here.